### PR TITLE
Add context menu and set to default option

### PR DESCRIPTION
**Pull request by gassmoeller**
Now with a right-click on an item it can be reset to its default value.

### CODE REVIEW

**Comment by tjhei** [comment]
:+1:

File: mainwindow.cpp [hunk 1]
```diff
@@ -94,6 +99,14 @@ namespace dealii
 
 
 
+    void MainWindow::set_to_default()
+    {
+      QTreeWidgetItem * current_item = tree_widget->currentItem();
+      current_item->setText(1,current_item->text(2));
```
Comment (bangerth) [inline mainwindow.cpp:105]: With Qt, who is responsible for eventually deleting this (and all of the other objects you create)?

Comment (gassmoeller) [reply mainwindow.cpp:105]: Here I just retrieve a pointer to an existing item that belongs to the tree_widget. But in general every Widget can have a parent (With the outermost parent being the main window). I suppose, but have not checked, that every parent is responsible for deleting its children. That being said, I am not sure all of our widgets specify a parent, and dont know what happens if they use the default (*parent=0). Do you want to open an issue for this?

Comment (bangerth) [reply mainwindow.cpp:105]: Yes, I chose the place to ask my question poorly -- but there are all of the other places where you/the previous author create objects with `new`.

No need for an issue. I was just curious. If we create a memory leak, then that's aesthetically not pleasing, but I'm not overly concerned about it in the current program.

**Pull request merged**